docs/SSR.md

```diff
@@ -0,0 +1,171 @@
+# Server Side Rendering (SSR)
+
+[<< Back to index](./index.md)
+
+The FSXA-PWA project is based on the [Nuxt.js](https://nuxtjs.org) framework which supports server side rendering ([SSR](https://ssr.vuejs.org/#what-is-server-side-rendering-ssr)). In a single page application which uses client side rendering (CSR) the webserver only serves an almost empty HTML page and usually larger amounts of javascript. The javascript then gets executed in the browser where all the required json data gets requested. The application then renders the view using the requested data. The SSR approach is different in that the json data gets requested from the webserver and the rendering of the view takes place there. A fully rendered page is then sent to the client. This allows for search engines to crawl the pages more easily since they do not have the necessity to be able to execute javascript code. It also saves time on the initial download of the page because there is no additional code execution step in between downloads. Nuxt.js uses a hybrid approach where the generated javascript bundle is able to hydrate the client (ie. request json data and render on the client side) from this point onwards.
```

Comment:
You typically spell JSON in capitals and JavaScript in "CamelCase" (several occurances in this paragraph).